metadata.go

```diff
@@ -14,11 +13,46 @@ type count struct {
 	Cnt int `json:"cnt"`
 }
 
+type metadataOptions struct {
+	tickerDuration time.Duration
+	awaitTimeout   time.Duration
+}
+
+type metadataOption func(*metadataOptions)
+
 // MetadataService is a service that runs druid metadata requests using druid SQL API.
 type MetadataService struct {
-	client             *Client
-	tickerMilliseconds time.Duration
-	waitTimeoutSeconds int
+	client         *Client
+	tickerDuration time.Duration
+	awaitTimeout   time.Duration
+}
+
+func NewMetadataService(client *Client, options []metadataOption) *MetadataService {
```

Comment:
Shall we also use var args here?

WDYT?

Reply:
Switched to var args.